time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -338,11 +418,19 @@ static CsvParsingContext readCsvHeader(CsvListReader csvListReader, String separ
         if (!duplicates.isEmpty()) {
             throw new TimeSeriesException("Bad CSV header, there are duplicates in time series names " + duplicates);
         }
-        List<String> names = Arrays.asList(tokens).subList(2, tokens.length);
-        return new CsvParsingContext(names);
+        List<String> names = Arrays.asList(tokens).subList(csvParserConfig.versioned() ? 2 : 1, tokens.length);
+        return new CsvParsingContext(names, csvParserConfig);
+    }
+
+    static void checkCsvHeader(String separatorStr, boolean versioned, String[] tokens) {
```

Comment:
I wonder if the separator should be added in the CsvParserConfig and then pass the config to this method

Reply:
Done